src/Microsoft.AspNetCore.Blazor/Components/BlazorComponent.cs

```diff
@@ -115,6 +116,22 @@ protected void StateHasChanged()
         protected virtual bool ShouldRender()
             => true;
 
+        /// <summary>
+        /// Method invoked each time after the component has been rendered.
+        /// </summary>
+        protected virtual void OnAfterRender()
+        {
+        }
+
+        /// <summary>
+        /// Method invoked each time after the component has been rendered. Note that the component does
+        /// not automatically re-render after the completion of any returned <see cref="Task"/>, because
+        /// that would cause an infinite render loop.
```

Comment:
If this is the case, is there a good reason to provide an async version of this? error handling?

Reply:
Sounds like you answered that already further down :)